tests/common_utils.py

```
@@ -0,0 +1,34 @@
+import json
+import sys
+from pathlib import Path
+
+
+def is_interactive():
+    return hasattr(sys, "ps1")
+
+
+def get_root():
+    if is_interactive():
+        return Path.cwd().parents[1].resolve()
+    else:
+        return Path(__file__).parents[1].resolve()
+
+
+def is_bad(path: Path, key: str):
+    root = get_root()
+    wdl_path = root / f"{path}/{path}.wdl"
+    conds = conditions(wdl_path)
+    return conds.get(key, False)
+
+
+def conditions(wdl_path: Path):
+    conditions_path = wdl_path.parent / "conditions.json"
+    if conditions_path.exists():
+        with open(conditions_path, "r") as f:
+            conditions = json.load(f)
+    else:
```

All - two points here:

1. This is where we set the default values of the conditions file if the file does not exist
2. The commented out `raise FileNotFoundError` block of code is to see if you like idea of raising an exception if there is no conditions file - that is, should all WDL dirs have a conditions.json file (in which case we can raise an error to make it clear it's missing), or do you want it to be optional (in which case we won't raise an error)

---

I am okay with it being optional.

---

thankis

---

What happens if the conditions file is present but there are no values?

---

I also agree with @seankross that this should be optional.

---

@sitapriyamoorthi  

> What happens if the conditions file is present but there are no values?

like it's a fully empty file, or just has an empty dictionary like `{}`? are both or just one of those likely in your mind?

---

100% agree that it should be optional. And in the case of an empty file, we can just use the default values for any missing parameters.

---

sound good